fe/fe-core/src/main/java/com/starrocks/common/util/RuntimeProfile.java

```diff
@@ -123,9 +122,11 @@ public Counter addCounter(String name, TUnit type, String parentName) {
             Counter newCounter = new Counter(type, 0);
             this.counterMap.put(name, Pair.create(newCounter, parentName));
 
-            TreeSet<String> childNames = childCounterMap.getOrDefault(parentName, new TreeSet<>());
+            if (!childCounterMap.containsKey(parentName)) {
+                childCounterMap.putIfAbsent(parentName, Sets.newConcurrentHashSet());
+            }
```

Comment:
Is it possible that two threads try to call `addCounter` with the same `parentName`?

Reply:
Maybe use `ConcurrentHashSet#computeIfAbsent`?
```java
Set<String> childNames = childCounterMap.computeIfAbsent(parentName, k -> Sets.newConcurrentHashSet());
```

Reply:
I think it is impossible

Reply:
/**
 * It is accessed by two kinds of thread, one is to create this RuntimeProfile
 * , named 'query thread', the other is to call
 * {@link com.starrocks.common.proc.CurrentQueryInfoProvider}.
 */

Reply:
Oh, maybe it is impossible.
Only coordinator calls it, and `Coordinate#updateProfile` has protected by `synchronized`.

Reply:
It is concurrent safe to do so, one of `ConcurrentHashMap::putIfAbsent` success and the others fail, with no side effect.